device/core/src/blob_upload/blob_upload_client.ts

```diff
@@ -27,15 +27,15 @@ export interface UploadParams {
  * @private
  */
 export interface FileUpload {
-    getBlobSharedAccessSignature(blobName: string, auth: X509 | SharedAccessSignature, done: (err: Error, uploadParams?: UploadParams) => void): void;
-    notifyUploadComplete(correlationId: string, auth: X509 | SharedAccessSignature, uploadResult: BlobUploadResult, done: (err?: Error) => void): void;
+  getBlobSharedAccessSignature(blobName: string, done: (err: Error, uploadParams?: UploadParams) => void): void;
+  notifyUploadComplete(correlationId: string, uploadResult: BlobUploadResult, done: (err?: Error) => void): void;
 }
 
 /**
  * @private
  */
 export interface BlobUploader {
```

Comment:
>BlobUploader [](start = 17, length = 12)

nit/gripe: these interfaces are hard to understand.  I think some of the interfaces get exposed to the device client, i think some of them get exposed to the end user (maybe), and I think some of them are used internally to the file/blob upload code, but I'm not completely sure.  Having 2 interfaces, one called BlobUpload and one called  BlobUploader that look almost the same also confuses me.  I know one is internal and one is external, but i'll be damned if I can figure out which.  I think it might benefit readability to add prefixes like "ProviderInterface" for internal interfaces that provide functionality for other code,  "ClientInterface" for things that outside users would see.  Maytbe.  I just think the interface name needs to describe what the interface does and not just the feature that it's involved with.  </soapbox>

Reply:
There's also some kind of relationship between FileUpload and BlobUpload, and i think maybe one uses the other, but this is also fuzzy

---
In reply to: [157072624](https://github.com/Azure/azure-iot-sdk-node/pull/180#discussion_r157072624) [](ancestors = 157072624)

Reply:
The point is, I don't think there's any breaking changes in this folder, but I honestly don't know because I can't figure out how the pieces fit together. (I mean, I could, but like all developers, I'm lazy and it looks like spaghetti to me.)

---
In reply to: [157072902](https://github.com/Azure/azure-iot-sdk-node/pull/180#discussion_r157072902) [](ancestors = 157072902,157072624)

Reply:
point taken. it's all old design though, not new stuff, so let's file this as part of future refactoring

---
In reply to: [157073502](https://github.com/Azure/azure-iot-sdk-node/pull/180#discussion_r157073502) [](ancestors = 157073502,157072902,157072624)